starport/services/network/blockchain.go

```diff
@@ -94,24 +104,89 @@ func (b *Blockchain) Init(ctx context.Context) error {
 	}
 
 	// cleanup home dir of app if exists.
-	if err := xos.RemoveAllUnderHome(chainHome); err != nil {
+	if err := os.RemoveAll(chainHome); err != nil {
 		return err
 	}
 
+	// build the chain and initialize it with a new validator key
+	b.builder.ev.Send(events.New(events.StatusOngoing, "Building the blockchain"))
 	if _, err := b.chain.Build(ctx, ""); err != nil {
 		return err
 	}
-
+	b.builder.ev.Send(events.New(events.StatusDone, "Blockchain built"))
+	b.builder.ev.Send(events.New(events.StatusOngoing, "Initializing the blockchain"))
 	if err := b.chain.Init(ctx, false); err != nil {
 		return err
 	}
-
 	b.builder.ev.Send(events.New(events.StatusDone, "Blockchain initialized"))
+
+	// write the custom genesis if a genesis URL is provided
+	if b.genesisURL != "" {
```

Comment:
Instead of duplicating can we reuse this part for publish and init by moving the logic to another func?